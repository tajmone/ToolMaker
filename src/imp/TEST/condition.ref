  OK 1
  OK 2
  OK 3
  OK 4
  OK 5
  OK 6
  OK 7
  OK 8
  OK 9
  OK 10
  OK 11
  OK 12
  OK 13
  OK 14
  OK 15
  OK 16
  OK 17
  OK 18
  OK 19
  OK 20
  OK 21
  OK 23
  OK 24
%%IF (5 = D[6])
%%IF (5 = D[6])                                            * * * FACIT * * *
%%ENDIF
%%IF (D[6] # D[6])
%%IF (D[6] # D[6])                                         * * * FACIT * * *
%%ENDIF
%%IF ({1,2,3}[5] < 7)
%%IF ({1,2,3}[5] < 7)                                      * * * FACIT * * *
%%ENDIF
%%IF (UNDEF*UNDEF[7] > 3)
%%IF (UNDEF*UNDEF[7] > 3)                                  * * * FACIT * * * 
%%ENDIF
%%IF (UNDEF <= 100)
%%IF (UNDEF <= 100)                                        * * * FACIT * * *
%%ENDIF
%%IF (UNDEF >= UNDEF)
%%IF (UNDEF >= UNDEF)                                      * * * FACIT * * *
%%ENDIF
%%IF (UNDEF > 3)
%%IF (UNDEF > 3)                                           * * * FACIT * * *
%%ELSE
%%ELSE                                                     * * * FACIT * * *
%%ENDIF
%%IF ({1, 2, 3, 4, 5, 6, 7, 8, 9} (%%(VAR15), %%(VAR14)))
%%IF ({1, 2, 3, 4, 5, 6, 7, 8, 9} (%%(VAR15), %%(VAR14)))  * * * FACIT * * *
%%ENDIF
  OK 25
  OK 26
