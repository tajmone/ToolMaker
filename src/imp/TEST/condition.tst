%%SET A ()
%%SET B (1)
%%SET C (1,2)
%%SET D (1,2,3)
%%SET E (1,2,3)
%%IF (C[2] = D[2])
  OK 1
%%ELSE
  Fel 1
%%ENDIF
%%IF (C[2] = D[3])
  Fel 2
%%ELSE
  OK 2
%%ENDIF
%%IF (C[2] # D[3])
  OK 3
%%ELSE
  Fel 3
%%ENDIF
%%IF (C[2] # D[2])
  Fel 4
%%ELSE
  OK 4
%%ENDIF
%%IF (C = D)
  Fel 5
%%ELSE
  OK 5
%%ENDIF
%%IF (E = D)
  OK 6
%%ELSE
  Fel 6
%%ENDIF
%%IF ({1,2,3} (%%(D)) AND A (?) OR {1,2,3,4,5,6} (%%(D)) AND {1,3,5}[0] = D[3])
  OK 7
%%ELSE
  Fel 7
%%ENDIF
%%IF ({3,1,2}[2] < {1,2,3,4,5,6}[2])
  OK 8
%%ELSE
  Fel 8
%%ENDIF
%%IF (C[2] < 2)
  Fel 9
%%ELSE
  OK 9
%%ENDIF
%%IF (2 > C[2])
  Fel 10
%%ELSE
  OK 10
%%ENDIF
%%IF (5 > D[3])
  OK 11
%%ELSE
  Fel 11
%%ENDIF
%%IF ({3,1,2}[3] <= {1,2,3,4,5,6}[2])
  OK 12
%%ELSE
  Fel 12
%%ENDIF
%%IF (C[2] <= 2)
  OK 13
%%ELSE
  Fel 13
%%ENDIF
%%IF (2 >= C[2])
  OK 14
%%ELSE
  Fel 14
%%ENDIF
%%IF (5 >= D[3])
  OK 15
%%ELSE
  Fel 15
%%ENDIF
%%SET HEJ (HOPP)
%%SET KALLE (HOPP)
%%IF (HEJ(HOPP))
  OK 16
%%ELSE
  Fel 16
%%ENDIF
%%IF (HEJ(KALLE))
  Fel 17
%%ELSE
  OK 17
%%ENDIF
%%SET VAR1 (1,2)
%%SET VAR2 (1,2)
%%SET VAR3 (2,1)
%%SET VAR4 (1,3)
%%SET VAR5 (3,2)
%%IF (VAR1 = VAR2)
  OK 18
%%ELSE
  Fel 18
%%ENDIF
%%IF (VAR1 = VAR4)
  Fel 19
%%ELSE
  OK 19
%%ENDIF
%%IF (VAR1 = VAR5)
  Fel 20
%%ELSE
  OK 20
%%ENDIF
%%IF (VAR1(%%(VAR3)) AND VAR3(%%(VAR1)))
  OK 21
%%ELSE
  Fel 22
%%ENDIF
%%IF (VAR1(%%(VAR1)))
  OK 23
%%ELSE
  Fel 23
%%ENDIF
%%SET VAR11 (1,2,3,4,5,6,7,8,9)
%%SET VAR12 (3,4)
%%SET VAR13 (5,6,7,8,9)
%%IF (VAR11 (1, 2, %%(VAR12), %%(VAR13)))
  OK 24
%%ELSE
  Fel 24
%%ENDIF
%%-- Partiell evaluaering
%%IF (5 = D[6])
`%%IF (5 = D[6])                                            * * * FACIT * * *
%%ENDIF
%%IF (D[6] # D[6])
`%%IF (D[6] # D[6])                                         * * * FACIT * * *
%%ENDIF
%%IF ({1,2,3}[5] < 7)
`%%IF ({1,2,3}[5] < 7)                                      * * * FACIT * * *
%%ENDIF
%%IF (1*UNDEF*UNDEF[7] > {1,2,3}[0])
`%%IF (UNDEF*UNDEF[7] > 3)                                  * * * FACIT * * * 
%%ENDIF
%%IF (UNDEF <= 94+C[2]+2*D[2])
`%%IF (UNDEF <= 100)                                        * * * FACIT * * *
%%ENDIF
%%IF (UNDEF >= UNDEF)
`%%IF (UNDEF >= UNDEF)                                      * * * FACIT * * *
%%ENDIF
%%IF (1=2)
  Fel 
%%ELSIF (UNDEF > 3)
`%%IF (UNDEF > 3)                                           * * * FACIT * * *
%%ELSIF (3 = 4)
  Fel
%%ELSIF (1=1)
`%%ELSE                                                     * * * FACIT * * *
%%ELSIF (C (D))
  Fel
%%ELSE
  Fel
%%ENDIF
%%IF (VAR11 (1,%%(VAR15), 2, %%(VAR14), %%(VAR13)))
`%%IF ({1, 2, 3, 4, 5, 6, 7, 8, 9} (`%%(VAR15), `%%(VAR14)))  * * * FACIT * * *
%%ENDIF
%%SET VAR15 (3)
%%SET VAR14 (4)
%%IF ({3,4} (1,%%(VAR15),2,%%(VAR14),5,6,7,8,9))
  Fel 25
%%ELSE
  OK 25
%%ENDIF
%%IF ({1,%%(VAR15),2,%%(VAR14),5,6,7,8,9} (3,4))
  OK 26
%%ELSE
  Fel 26
%%ENDIF
