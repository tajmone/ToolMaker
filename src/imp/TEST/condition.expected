########## condition ##########
  OK 1
  OK 2
  OK 3
  OK 4
  OK 5
  OK 6
  OK 7
  OK 8
  OK 9
  OK 10
  OK 11
  OK 12
  OK 13
  OK 14
  OK 15
  OK 16
  OK 17
  OK 18
  OK 19
  OK 20
  OK 21
  OK 23
  OK 24
%%IF (5 = D[6])
%%IF (5 = D[6])                                            * * * FACIT * * *
%%ENDIF
%%IF (D[6] # D[6])
%%IF (D[6] # D[6])                                         * * * FACIT * * *
%%ENDIF
%%IF ({1,2,3}[5] < 7)
%%IF ({1,2,3}[5] < 7)                                      * * * FACIT * * *
%%ENDIF
%%IF (UNDEF*UNDEF[7] > 3)
%%IF (UNDEF*UNDEF[7] > 3)                                  * * * FACIT * * * 
%%ENDIF
%%IF (UNDEF <= 100)
%%IF (UNDEF <= 100)                                        * * * FACIT * * *
%%ENDIF
%%IF (UNDEF >= UNDEF)
%%IF (UNDEF >= UNDEF)                                      * * * FACIT * * *
%%ENDIF
%%IF (UNDEF > 3)
%%IF (UNDEF > 3)                                           * * * FACIT * * *
%%ELSE
%%ELSE                                                     * * * FACIT * * *
%%ENDIF
%%IF ({1, 2, 3, 4, 5, 6, 7, 8, 9} (%%(VAR15), %%(VAR14)))
%%IF ({1, 2, 3, 4, 5, 6, 7, 8, 9} (%%(VAR15), %%(VAR14)))  * * * FACIT * * *
%%ENDIF
  OK 25
  OK 26


condition.tst

  132.  %%IF (5 = D[6])
=====>              1

  *1*   209 E : Index value "6" out of range.

  135.  %%IF (D[6] # D[6])
=====>          1      2

  *1*   209 E : Index value "6" out of range.
  *2*   209 E : Index value "6" out of range.

  138.  %%IF ({1,2,3}[5] < 7)
=====>                1

  *1*   209 E : Index value "5" out of range.

  141.  %%IF (1*UNDEF*UNDEF[7] > {1,2,3}[0])
=====>          1     2

  *1*   204 W : Variable "UNDEF" not set.
  *2*   204 W : Variable "UNDEF" not set.

  144.  %%IF (UNDEF <= 94+C[2]+2*D[2])
=====>        1

  *1*   204 W : Variable "UNDEF" not set.

  147.  %%IF (UNDEF >= UNDEF)
=====>        1        2

  *1*   204 W : Variable "UNDEF" not set.
  *2*   204 W : Variable "UNDEF" not set.

  152.  %%ELSIF (UNDEF > 3)
=====>           1

  *1*   204 W : Variable "UNDEF" not set.

  163.  %%IF (VAR11 (1,%%(VAR15), 2, %%(VAR14), %%(VAR13)))
=====>                    1             2

  *1*   204 W : Variable "VAR15" not set.
  *2*   204 W : Variable "VAR14" not set.


4 error(s) detected.
8 warning(s) issued.

